libcontainerd/client_daemon.go

```diff
@@ -639,6 +643,14 @@ func (c *client) processEvent(ctr *container, et EventType, ei EventInfo) {
 			c.Lock()
 			delete(ctr.execs, ei.ProcessID)
 			c.Unlock()
+			ctr := c.getContainer(ei.ContainerID)
```

Comment:
On error (e.g.  this would cause a deadlock if I'm not mistaken

Reply:
Can you explain more? You mean the `RLock` inside `getContainer()` ?

Reply:
Yes, I think I had this initially (a call to `getContainer`) and it caused a deadlock in some error conditions and for execs. Can't remember the exact details

Reply:
Well, it is taking the same lock as line 639 so I'm not sure how this can cause a deadlock and the previous one doesn't.

Reply:
I may just be confused, I'll have to have a closer look, I had to change this part quite a few time to get rid of all the deadlocks. I'll try to have a proper look sometime tomorrow.

Reply:
Just ignore me, this is safe :innocent: I got confused with another part of the code.